ddp-workspace/projects/toolkit/src/lib/components/dashboard/dashboard-redesigned.component.ts

```diff
@@ -258,28 +246,20 @@ export class DashboardRedesignedComponent extends DashboardComponent implements
             });
     }
 
-    public removeParticipant(userGuid): void {
-        this.participantToRemoveButtonDisabled.set(userGuid, true);
-        this.userManagementService.deleteUser(userGuid)
-            .pipe(finalize(() => this.participantToRemoveButtonDisabled.set(userGuid, false)))
-            .subscribe(() => {
-                this.removedParticipants$.next([...this.removedParticipants$.value, userGuid]);
-            });
-    }
+    private checkAndDeleteAccidentalParticipant(participants: DashboardParticipant[]): Observable<DashboardParticipant[]> {
+        const accidentalParticipant = participants.filter(({activities}) => !activities.length);
 
-    public isParticipantOperator(participantGuid: string): boolean {
-        return participantGuid === this.session.session?.userGuid;
+        if (!accidentalParticipant.length) {
+            return of(participants);
+        }
+
+        return forkJoin(...accidentalParticipant.map((participant) => this.userManagementService.deleteUser(participant.userGuid)))
+            .pipe(map(() => participants.filter(({activities}) => activities.length)));
```

Comment:
I updated the code a little bit, hope it's more clear now

Reply:
OK. 2 comments:
1. Maybe `mapTo` instead of `map` . I think a little more apropos to what you are doing.
2. Could you double check that we can count on deleteUser not generating an exception? If it does, we need to handle somehow here. Just that user deletion has always been touchy in our system. I am seeing, currently that we should worry about http codes 422, 401, and 404.

Reply:
done! I also fixed the generating custom label for chilfren w/o names and moved it after empty users are deleting in order to put the correct index number into child title